File: Advent of Code/2015-Day4.exs
# Day4 - Elixir
# Xophidia
# based on Mark Olson's code

defmodule Crypto do
    def md5(data) do
        Base.encode16(:erlang.md5(data), case: :lower)
    end
end

defmodule Launcher do
    def isValid(data) do
        String.match?(Crypto.md5(data), ~r/^00000/)
    end

    def exec,do: exec(0)
    def exec(i) do
        key = "ckczppom"
        IO.puts i
        unless isValid("#{key}#{i}"), do: exec(i+1)
    end
end

Launcher.exec()

File: Advent of Code/2015-Day4.py
#!/usr/bin/env python
# AdventofCode - Day4
# xophidia

import md5
chaine = "0123456789"

for i in chaine:
	for j in chaine:
		for k in chaine:
			for l in chaine:
				for m in chaine:
					for n in chaine:
							temp = md5.new("ckczppom" + i + j + k + l + m + n).hexdigest()
							if temp[:5] == '00000':
								print temp
								print " %c%c%c%c%c%c " % (i,j,k,l,m,n)
								
								 